### ISSUE TITLE
Quantify the results

### ISSUE THREAD

**Issue by 05063112lcs** [TQTQliu/ET-MVSNet#16]
Hello, I'm sorry to bother you again, it may be that my current level is low, and when I read your paper, I found that the quantitative results on the DTU and TNT datasets you mentioned in the paper, for the test results of the TNT dataset, I can get them by running the two test files you provided? Or do you need to take additional steps, for the data in your paper, do you get it from Tensorboard?

**Comment by TQTQliu** [comment]
First of all, the goal of MVS is to reconstruct point clouds, which is a two-step process: 1) generate depth maps through a neural network (i.e., the model), 2) filter and fuse depth maps into point clouds. The quantitative results presented in the paper (Table 1 and Table. 2) are all metrics evaluated for reconstructed point clouds.

Therefore, if you want to get the quantitative results, you should: 
1) run the test script to get the reconstructed point clouds, for example, run the [test script](https://github.com/TQTQliu/ET-MVSNet/blob/e29d0b6c1d7772a86b730f4b8b106df3846a7fb2/README.md?plain=1#L105) for DTU dataset.
2) run [here](https://github.com/TQTQliu/ET-MVSNet/blob/e29d0b6c1d7772a86b730f4b8b106df3846a7fb2/README.md?plain=1#L116) to calculate the error between reconstructed point clouds and the Ground Truth. This is a Matlab-based evaluation code that all MVS methods use as a fair comparison. After running, the final quantitative results will be printed in the terminal, and will also be saved in `metric.txt` (the evaluation results for each scenes) and `overall.txt` (the average of the evaluation results for all scenes), which you can find in the corresponding output path. It is worth noting that this matlab-based code may take a long time to run (several hours), depending on different hardware conditions. More details can be found in `readme`. 

Hope it helps. Just feel free to ask questions~

**Comment by 05063112lcs** [comment]
Hello author, sorry to bother you so late! Today, I put my training results into TNT data set for testing according to what you wrote in reademe, but the final results only generated cams, mask, confidence and other relevant information of each scene under the output folder, and I did not find the txt file containing indicator information you mentioned. Is there something wrong with the way I operate? (The dtu data set is used as the training set, the trained model is placed in the blendedMVS data set for fine tuning, and the adjusted result is used as the checkpoint of TNT data set for testing)



---Original---
From: "Tianqi ***@***.***&gt;
Date: Fri, Jun 7, 2024 21:37 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [TQTQliu/ET-MVSNet] Quantify the results (Issue #16)




 
First of all, the goal of MVS is to reconstruct point clouds, which is a two-step process: 1) generate depth maps through a neural network (i.e., the model), 2) filter and fuse depth maps into point clouds. The quantitative results presented in the paper (Table 1 and Table. 2) are all metrics evaluated for reconstructed point clouds.
 
Therefore, if you want to get the quantitative results, you should:
  
run the test script to get the reconstructed point clouds, for example, run the test script for DTU dataset.
 
run here to calculate the error between reconstructed point clouds and the Ground Truth. This is a Matlab-based evaluation code that all MVS methods use as a fair comparison. After running, the final quantitative results will be printed in the terminal, and will also be saved in metric.txt (the evaluation results for each scenes) and overall.txt (the average of the evaluation results for all scenes), which you can find in the corresponding output path. It is worth noting that this matlab-based code may take a long time to run (several hours), depending on different hardware conditions. More details can be found in readme.
  
Hope it helps. Just feel free to ask questions~

**Comment by TQTQliu** [comment]
For DTU, it provides ground-truth point clouds, so that quantitative metrics can be obtained through matlab evaluation code. 
But for TNT, this is an online dataset, and the ground-truth point cloud is not public, so you need to upload the reconstructed point cloud (.ply) to [Tanks and Temples benchmark](https://www.tanksandtemples.org/) for evaluation. Refer [here](https://github.com/TQTQliu/ET-MVSNet/blob/e29d0b6c1d7772a86b730f4b8b106df3846a7fb2/README.md?plain=1#L130).
